source/common/http/http1/codec_impl.cc

```diff
@@ -261,6 +263,11 @@ void StreamEncoderImpl::endEncode() {
 
   connection_.flushOutput(true);
   connection_.onEncodeComplete();
+  // If half-close semantics are enabled, no chunk encoding nor content length
+  // are given, we signal end stream by half-closing the connection.
+  if (connection_.connection().isHalfCloseEnabled() && !chunk_encoding_) {
```

Comment:
I should add a check on `content_length`

Reply:
yes please.